tests/test_CompareEclipseRestart.cpp

```diff
@@ -0,0 +1,173 @@
+                std::cout << "Median relative deviation = " << ReadUNRST::median(relDeviation) << std::endl;
+            }
+            std::cout << std::endl;
+            read.close();
+        }
+    }
+    return 0;
+}
+
+//------------------------------------------------//
+
+void printHelp() {
+    std::cout << "The program takes four arguments:\n"
+        << "1. .EGRID-file number 1\n" 
+        << "2. .EGRID-file number 2\n" 
+        << "2. .UNRST-file number 1\n" 
+        << "3. .UNRST-file number 2\n";
+}
+
+bool ReadUNRST::open(const char* unrstFile1, const char* unrstFile2) {
+    if (ecl_file1 == NULL) {
+        ecl_file1 = ecl_file_open(unrstFile1, ECL_FILE_CLOSE_STREAM);
+    }
+    if (ecl_file2 == NULL) {
+        ecl_file2 = ecl_file_open(unrstFile2, ECL_FILE_CLOSE_STREAM);
+    }
+
+    if (ecl_file1 == NULL || ecl_file2 == NULL) {
+        return false;
+    }
+    else {
+        return true;
+    }
+}
+
+void ReadUNRST::close() {
+    ecl_file_close(ecl_file1);
+    ecl_file_close(ecl_file2);
+}
+
+bool ReadUNRST::results(const char* keyword, std::vector<double>* absDeviation, std::vector<double>* relDeviation) {
+    if (!ecl_file_has_kw(ecl_file1, keyword) || !ecl_file_has_kw(ecl_file2, keyword)) {
+        std::cout << "The file does not have this keyword." << std::endl;
+        return false;
+    }
+    unsigned int occurrences1 = ecl_file_get_num_named_kw(ecl_file1, keyword);
+    unsigned int occurrences2 = ecl_file_get_num_named_kw(ecl_file2, keyword);
+    if (occurrences1 != occurrences2) {
+        std::cout << "Keyword occurrences are not equal." << std::endl;
+        return false;
+    }
+
+    for (unsigned int index = 0; index < occurrences1; ++index) {
+        ecl_kw_type* ecl_kw1 = ecl_file_iget_named_kw(ecl_file1, keyword, index);
+        unsigned int numActiveCells1 = ecl_kw_get_size(ecl_kw1);
+        ecl_kw_type* ecl_kw2 = ecl_file_iget_named_kw(ecl_file2, keyword, index);
+        unsigned int numActiveCells2 = ecl_kw_get_size(ecl_kw2);
+        if (numActiveCells1 != numActiveCells2) {
+            std::cout << "Number of active cells are different." << std::endl;
+            return false;
+        }
+
+        std::vector<double> values1; // Elements in the vector corresponds to active cells
+        std::vector<double> values2; // Elements in the vector corresponds to active cells
+        values1.resize(numActiveCells1);
+        values2.resize(numActiveCells2);
+        ecl_kw_get_data_as_double(ecl_kw1, values1.data());
+        ecl_kw_get_data_as_double(ecl_kw2, values2.data());
+
+        for (unsigned int i = 0; i < values1.size(); i++) {
+            calculateDeviations(absDeviation, relDeviation, values1[i], values2[i]);
+        }
+    }
+    return true;
+}
+
+
+void ReadUNRST::calculateDeviations(std::vector<double> *absDeviationVector,std::vector<double> *relDeviationVector, double val1, double val2){
```

Comment:
The management of storing the numbers should be separated from the computation.

It would be much easier to understand this method if it only took val1 and val2 as arguments, and returned a pair of doubles (the abs and rel deviation) or a tiny custom struct even (struct Deviation { double abs; double rel; }).

Then the loop above (line 114) should push to the vectors instead of that happening in calculateDeviations.